gazebo_ros/test/plugins/sdf_node.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/common/Plugin.hh>
+
+#include <std_msgs/msg/string.hpp>
+#include <gazebo_ros/node.hpp>
+
+#include <memory>
+
+/// Simple example of a gazebo system plugin which uses a ROS2 node with gazebo_ros::Node.
+class SDFNode : public gazebo::WorldPlugin
+{
+public:
+  /// Called by gazebo to load plugin. Creates #node, #timer_, and #pub
+  /// \param[in] argc Argument count.
+  /// \param[in] argv Argument values.
+  void Load(gazebo::physics::WorldPtr world, sdf::ElementPtr sdf);
+
+private:
+  /// Timer called to publish a message every second
+  std::shared_ptr<rclcpp::TimerBase> timer_;
+};
+
+void SDFNode::Load(gazebo::physics::WorldPtr, sdf::ElementPtr _sdf)
+{
+  // It should be ok to create a node without calling init first.
+  auto node = gazebo_ros::Node::Create("sdf_node", _sdf);
+  assert(nullptr != node);
+
+  const char * node_name = node->get_name();
+  if (strcmp("my_node", node_name)) {
+    RCLCPP_ERROR(node->get_logger(), "Node name not remapped in SDF");
+    return;
+  }
+
+  // Create a publisher
+  auto pub = node->create_publisher<std_msgs::msg::String>("test");
+
+  // Attempt to get the parameters set by the SDF.
+  // If any of them fail, an exception will be thrown causing the test to fail
```

Comment:
It looks like it doesn't really throw an exception. The test times out because the plugin never publishes a message. We may want to change this comment.